prusti-viper/src/encoder/snapshot_encoder.rs

```diff
@@ -183,6 +183,7 @@ impl<'p, 'v, 'r: 'v, 'a: 'r, 'tcx: 'a> SnapshotEncoder<'p, 'v, 'tcx> {
                     unreachable!()
                 }
             }
+            // Refs should be stripped by now by [Encoder::encode_snapshot]
             x => unimplemented!("{:?}", x),
```

Comment:
(Not for this PR) at some point, we should replace this `unimplemented` with an  `EncodingError::internal` in case `x` is a reference and with `EncodingError::unsupported` in all the remaining cases.

Reply:
The case should actually never be reached because this is only invoked after the snapshot check for `is_ty_supported`. However, `is_ty_supported` returns `true` for `Ref`s (again because they are handled in `Encoder::encode_snapshot`, so I added an explanation for why they don't appear here.